api/python/quilt3/packages.py

```diff
@@ -668,28 +671,41 @@ def load(cls, readable_file):
 
     @classmethod
     def _load(cls, readable_file):
-        reader = jsonlines.Reader(readable_file)
-        meta = reader.read()
-        meta.pop('top_hash', None)  # Obsolete as of PR #130
-        pkg = cls()
-        pkg._meta = meta
-        for obj in reader:
-            path = cls._split_key(obj.pop('logical_key'))
-            subpkg = pkg._ensure_subpackage(path[:-1])
-            key = path[-1]
-            if not obj.get('physical_keys', None):
-                # directory-level metadata
-                subpkg.set_meta(obj['meta'])
-                continue
-            if key in subpkg._children:
-                raise PackageException("Duplicate logical key while loading package")
-            subpkg._children[key] = PackageEntry(
-                obj['physical_keys'],
-                obj['size'],
-                obj['hash'],
-                obj['meta']
-            )
 
+        gc.disable()  # Experiments with COCO (650MB manifest) show disabling GC gives us ~2x performance improvement
+
+        line_count = 0
+        for _ in readable_file:
```

Comment:
That seems pretty inefficient. Just count bytes instead? See `OSUtils().open_file_chunk_reader(...)` in `data_transfer.py`

Reply:
I've found that it is a very small portion of the call so I think it's fine (on my laptop, `cv/coco2017` ~0.7 seconds to scan the line count compared to 70 seconds to download the manifest and 10-11 seconds to parse the jsonl) so I think it's probably fine. The `jsonlines` reader doesn't give a way for us to see how many bytes each parsed json is.

Reply:
Ok. That's fine I guess.

(But you don't need any support from `jsonlines` - you only need to wrap `readable_file` in order to count bytes.)